Fix for navbar inputs with Floating Labels.

wallter opened the pull request:
# Before:
<img width="738" alt="b1" src="https://cloud.githubusercontent.com/assets/1613806/13296433/c36d2f66-daea-11e5-8a52-c5351f7b47d9.png">
<img width="731" alt="b2" src="https://cloud.githubusercontent.com/assets/1613806/13296432/c36c2026-daea-11e5-8b6f-a6796be20d76.png">
<img width="493" alt="b3" src="https://cloud.githubusercontent.com/assets/1613806/13296431/c36b2004-daea-11e5-9a03-dc9709293f67.png">
<img width="498" alt="b4" src="https://cloud.githubusercontent.com/assets/1613806/13296430/c36b0c4a-daea-11e5-8580-f6f797fce0c7.png">


# After:
<img width="730" alt="a1" src="https://cloud.githubusercontent.com/assets/1613806/13296440/c9928148-daea-11e5-924a-7474d82cc735.png">
<img width="737" alt="a2" src="https://cloud.githubusercontent.com/assets/1613806/13296441/c99f4658-daea-11e5-8141-d781270f8679.png">
<img width="494" alt="a3" src="https://cloud.githubusercontent.com/assets/1613806/13296442/c9a4c9de-daea-11e5-8aa3-dd6beebc25c7.png">
<img width="493" alt="a4" src="https://cloud.githubusercontent.com/assets/1613806/13296443/c9a5f73c-daea-11e5-8b8f-8b39c2c68171.png">

rosskevin commented on .gitignore:
```diff
@@ -15,7 +15,7 @@ Thumbs.db
 .DS_Store
 /node_modules/
 .grunt/
-/bower_components/
```
Please revert this file

rosskevin commented on dist/css/bootstrap-material-design.css:
```diff
@@ -2334,24 +2334,54 @@ fieldset[disabled] .navbar .btn-link:focus {
   background-color: #009688;
   color: rgba(255,255,255, 0.84);
 }
-.navbar .navbar-form .form-group input.form-control::-moz-placeholder,
```
Please revert all dist files, they cloud actual changes.

rosskevin commented on less/_navbar.less:
```diff
@@ -188,13 +188,47 @@
+            text-overflow: ellipsis;
+            overflow: hidden;
+            white-space: nowrap;
+            opacity: 0.87;
```
The majority of this change should not be inside a .generic-variations call.  This creates 4 or so color variations.  If it isn't overriding the color, it should not be in this file.

rosskevin commented on less/_navbar.less:
```diff
@@ -188,13 +188,47 @@
+            }
+          }
+        }
+      }
```
any change of the variations needs to mirror the change in the `scss` hierarchy in the relevant scss mixin.  There is usually a comment referencing the specific one.  That's another reason (other than 4 copies) to keep only overrides inside it.

wallter commented:
@rosskevin - I am not that familiar to less and even less so with `scss`. I would welcome being showed how to rewrite of the few rules I've added - in both less and scss?

I've reverted the `dist/*` files as well as updated the `.gitignore` per your recommendation.

rosskevin commented:
First, thank you for taking the time to submit a PR.  I encourage you to look further at the less and sass sites, you'll find that they are very much the same, and the nuances are already separated in the code.  Unfortunately, I don't have time to help you further, but you could submit your questions to stackoverflow and reference this pull.  I don't think you are far away from the solution.
